fn/either_test.go

```diff
@@ -0,0 +1,139 @@
+package fn
+
+import (
+	"testing"
+	"testing/quick"
+)
+
+func TestPropConstructorEliminatorDuality(t *testing.T) {
+	f := func(i int, s string, isRight bool) bool {
+		Len := func(s string) int { return len(s) } // smh
```

Comment:
Q: what does the comment mean smh ?

Reply:
"Shake my head", means disappointment. I was making some commentary about how I couldn't just put `len` in the field as the type `func(string) int` even though it would be a sensible thing to be able to do. `append` has similar issues. Perhaps I should either remove the snarky comment or be more explicit about warning people not to try and refactor it away.